### ISSUE TITLE
custom foreign_key replaced with associated records primary key in belongs_to

### ISSUE THREAD

**Issue by marcgreenstock** [zdennis/activerecord-import#469]
I discovered after #448 that a model persisted with import that has a custom foreign_key in a belongs_to association is being replaced with the primary key of the associated record.

```ruby
class Category < ApplicationRecord
  has_many :products, foreign_key: :ref_id, primary_key: :ref_id
end

class Product < ApplicationRecord
  belongs_to :category, foreign_key: :ref_id, primary_key: :ref_id
end
```

Given a category row with:

| id | ref_id |
| -- | ----------- |
| 5bffa9e2-cd56-471e-b69b-8773da1d7b4f | ref_456 | 

Before #448, `Product.import` would save the product record with:

| id | ref_id |
| -- | ----------- |
| 5ca21e1e-36b1-47c2-a182-4bc7859b1980 | ref_456 |

Now it is being saved with

| id | ref_id |
| -- | ----------- |
| 5ca21e1e-36b1-47c2-a182-4bc7859b1980 | 5bffa9e2-cd56-471e-b69b-8773da1d7b4f |

This makes `product.category` return `nil` instead of the associated record.

The following line is the culprit:
```ruby
model.public_send("#{association_reflection.foreign_key}=", association.id)
```
I am wondering why was this introduced?

I am monkey patching with the following to disable this at the moment:

```ruby
class ActiveRecord::Base
  class << self
    def load_association_ids(_model); end
  end
end
```

**Comment by jkowens** [comment]
The reason for this is that if an array of parent models were imported and then you attempted to import one of their associations, the foreign key ID wasn't populated. Here is a scenario:

https://github.com/zdennis/activerecord-import/issues/402#issuecomment-328375171

I think I discovered where I went wrong and have created PR #470. If you are able to test it out that would be awesome. Thanks 👍

**Comment by jkowens** [comment]
This fix has been released in v0.21.0 🎉
